In [ ]:
!pip install -q git+https://github.com/THU-MIG/yolov10.git

  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done


In [ ]:
!pip install -U albumentations


   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 224.0/224.0 kB 10.4 MB/s eta 0:00:00
  Attempting uninstall: albucore
    Found existing installation: albucore 0.0.16
    Uninstalling albucore-0.0.16:
      Successfully uninstalled albucore-0.0.16
  Attempting uninstall: albumentations
    Found existing installation: albumentations 1.4.15
    Uninstalling albumentations-1.4.15:
      Successfully uninstalled albumentations-1.4.15


In [ ]:
!ls

sample_data


In [ ]:
!mkdir -p HOME/weights
!wget -P HOME/weights -q https://github.com/jameslahm/yolov10/releases/download/v1.0/yolov10n.pt
!wget -P HOME/weights -q https://github.com/jameslahm/yolov10/releases/download/v1.0/yolov10s.pt
!wget -P HOME/weights -q https://github.com/jameslahm/yolov10/releases/download/v1.0/yolov10m.pt
!wget -P HOME/weights -q https://github.com/jameslahm/yolov10/releases/download/v1.0/yolov10b.pt
!wget -P HOME/weights -q https://github.com/jameslahm/yolov10/releases/download/v1.0/yolov10x.pt
!wget -P HOME/weights -q https://github.com/jameslahm/yolov10/releases/download/v1.0/yolov10l.pt
!ls -lh HOME/weights

total 408M
-rw-r--r-- 1 root root  80M May 23 09:38 yolov10b.pt
-rw-r--r-- 1 root root 100M May 23 09:38 yolov10l.pt
-rw-r--r-- 1 root root  64M May 23 09:38 yolov10m.pt
-rw-r--r-- 1 root root  11M May 23 09:38 yolov10n.pt
-rw-r--r-- 1 root root  32M May 23 09:38 yolov10s.pt
-rw-r--r-- 1 root root 123M May 23 09:38 yolov10x.pt


In [ ]:
from ultralytics import YOLOv10

In [ ]:
!pip install roboflow

from roboflow import Roboflow
rf = Roboflow(api_key="hc3ubFnCASkRXDptHnwO")
project = rf.workspace("visionhub-vinro").project("truck-cm5du")
version = project.version(1)
dataset = version.download("yolov9")


   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 80.3/80.3 kB 4.8 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 66.8/66.8 kB 6.5 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.5/54.5 kB 4.4 MB/s eta 0:00:00
  Attempting uninstall: idna
    Found existing installation: idna 3.10
    Uninstalling idna-3.10:
      Successfully uninstalled idna-3.10
loading Roboflow workspace...
loading Roboflow project...



Extracting Dataset Version Zip to Truck-1 in yolov9:: 100%|██████████| 2374/2374 [00:00<00:00, 6006.92it/s]


In [ ]:
!pip install roboflow

from roboflow import Roboflow
rf = Roboflow(api_key="hc3ubFnCASkRXDptHnwO")
project = rf.workspace("visionhub-vinro").project("vehicles-images-dataset")
version = project.version(1)
dataset = version.download("yolov9")


In [42]:
!pip install roboflow

from roboflow import Roboflow
rf = Roboflow(api_key="LnzZeidf9sFEMC6MY6a3")
project = rf.workspace("salah-4hfpa").project("v-4dl4b")
version = project.version(1)
dataset = version.download("yolov9")


loading Roboflow workspace...
loading Roboflow project...



Extracting Dataset Version Zip to V-1 in yolov9:: 100%|██████████| 3942/3942 [00:01<00:00, 3327.06it/s]


In [ ]:
!git clone https://github.com/THU-MIG/yolov10.git

Cloning into 'yolov10'...
remote: Enumerating objects: 20332, done.
remote: Counting objects: 100% (2446/2446), done.
remote: Compressing objects: 100% (250/250), done.
remote: Total 20332 (delta 2317), reused 2197 (delta 2196), pack-reused 17886 (from 1)
Receiving objects: 100% (20332/20332), 11.14 MiB | 10.74 MiB/s, done.
Resolving deltas: 100% (14337/14337), done.


In [ ]:
!pwd

/content/yolov10


In [ ]:
cd yolov10/


/content/yolov10


In [ ]:
!pip install -qr requirements.txt

### **Fine-Tune the YOLO Model**

**Connecting to  Google Drive**

In [ ]:
from google.colab import drive
drive.mount('/content/drive')


Mounted at /content/drive


**Saving Weight to Google Drive**

In [ ]:
# Set your save directory
save_dir = '/content/drive/MyDrive/YOLO_weights'

In [ ]:
!yolo task=detect mode=train epochs=15 batch=8 plots=True model=/content/HOME/weights/yolov10n.pt data=/content/V-1/data.yaml --project=YOLO_weights --name=train_results --save=True

WARNING ⚠️ argument '--project=YOLO_weights' does not require leading dashes '--', updating to 'project=YOLO_weights'.
WARNING ⚠️ argument '--name=train_results' does not require leading dashes '--', updating to 'name=train_results'.
WARNING ⚠️ argument '--save=True' does not require leading dashes '--', updating to 'save=True'.
New https://pypi.org/project/ultralytics/8.3.18 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.1.34 🚀 Python-3.10.12 torch-2.0.1+cu117 CPU (Intel Xeon 2.20GHz)
engine/trainer: task=detect, mode=train, model=/content/HOME/weights/yolov10n.pt, data=/content/V-1/data.yaml, epochs=15, time=None, patience=100, batch=8, imgsz=640, save=True, save_period=-1, val_period=1, cache=False, device=None, workers=8, project=YOLO_weights, name=train_results4, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profil

**GPU_mem:**

This shows the amount of GPU memory being used during training. Lower values indicate more efficient memory usage, while higher values can indicate that you're close to your GPU's limits.

**box_om (Objectness Loss):**

This value represents the loss related to predicting the presence of an object. A **lower** value indicates that the model is better at detecting whether an object exists in a predicted bounding box.

**cls_om (Classification Loss):**

This measures how well the model is classifying detected objects. A **lower** classification loss suggests that the model is improving its ability to correctly classify objects within the detected boxes.

**dfl_om (Distribution Focal Loss):**

This metric is used in some YOLO implementations to measure the quality of the predicted bounding box coordinates. **Lower** values indicate better localization of objects.

**box_oo (Objectness Output):**

This metric reflects the objectness score for the predicted bounding boxes. **Higher** values indicate a stronger confidence that an object is present in the predicted bounding box.

**cls_oo (Classification Output):**

This shows the output classification scores for the detected objects. **Higher** values indicate better confidence in the predicted class for each detected object.

**dfl_oo (Distribution Focal Loss Output):**

Similar to dfl_om, this indicates the model's performance in predicting bounding box coordinates for output.

**Instances:**

This indicates the number of instances (or objects) detected in the current batch. It gives insight into how many objects the model is working with during training.

**Size:**

This shows the input size of the images being processed (e.g., 640 pixels). It indicates the dimensions used for training and affects the model's performance and accuracy.

Overall, these metrics help you gauge how well your model is training and if adjustments are needed to improve performance. If you have specific concerns about any of these values, let me know!




**Class**: Indicates the class being evaluated. In your case, it seems to be a single class, likely "truck."

**Images**:330: The total number of images used for evaluation.

**Instances**:

287: The total number of object instances (bounding boxes) that were present in the evaluation dataset.

**Box(P)**: This is the precision of the bounding boxes, representing the ratio of true positive detections to the total detections (true positives + false positives).  

**R (Recall)**: This indicates the recall, which measures the ratio of true positive detections to the total actual instances (true positives + false negatives).

**mAP50**: This is the mean Average Precision at IoU (Intersection over Union) threshold of 0.50. It reflects the model’s performance in detecting objects.

**mAP50-95**: This is the mean Average Precision calculated over multiple IoU thresholds (from 0.50 to 0.95). This gives a more comprehensive view of the model's performance across varying levels of detection accuracy.

**Training Duration:** 15 epochs took approximately 5.8 hours.
# **Model Performance:**
After the final epoch, achieved a precision of 0.924 and recall of 0.895 for the validation dataset, with a mean Average Precision (mAP50) of 0.951 and mAP50-95 of 0.929.
**Parameters:** The model has around 2.69 million parameters.

In [1]:
!pwd

/content


In [41]:
%cd ..

/content
